ext/storage_sources/dir_store/dir_store.c

```diff
@@ -1090,15 +1090,62 @@ dir_store_rename(WT_FILE_SYSTEM *file_system, WT_SESSION *session, const char *f
 
 /*
  * dir_store_remove --
- *     POSIX remove, not supported for cloud objects.
+ *     Calls the corresponding remove method of the file system behind our dir_store storage source
  */
 static int
 dir_store_remove(WT_FILE_SYSTEM *file_system, WT_SESSION *session, const char *name, uint32_t flags)
 {
-    (void)flags; /* unused */
+    DIR_STORE_FILE_SYSTEM *dir_store_fs;
+    size_t bucket_file_len, cache_file_len;
+    WT_FILE_SYSTEM *wt_fs;
+    bool existp;
+    int ret;
+    char *bucket_filename, *cache_filename;
 
-    return (dir_store_err(
-      FS2DS(file_system), session, ENOTSUP, "%s: remove of file not supported", name));
+    existp = false;
+    ret = 0;
+    dir_store_fs = (DIR_STORE_FILE_SYSTEM *)file_system;
+    wt_fs = dir_store_fs->wt_fs;
+
+    /* Skip over "./" and variations at the beginning of the name. */
+    while (*name == '.') {
+        if (name[1] != '/')
+            break;
+        name += 2;
+        while (*name == '/')
+            name++;
+    }
+
+    /* Generate the paths to the shared object files in the bucket and cache directories. */
+    bucket_file_len = strlen(dir_store_fs->bucket_dir) + strlen(name) + 2;
+    if ((bucket_filename = calloc(bucket_file_len, sizeof(char))) == NULL)
+        return (dir_store_err(FS2DS(file_system), session, ENOMEM, "dir_store_remove"));
+    if (snprintf(bucket_filename, bucket_file_len, "%s/%s", dir_store_fs->bucket_dir, name) >=
+      (int)bucket_file_len)
+        return (dir_store_err(FS2DS(file_system), session, EINVAL, "overflow sprintf"));
+
+    cache_file_len = strlen(dir_store_fs->cache_dir) + strlen(name) + 2;
+    if ((cache_filename = calloc(cache_file_len, sizeof(char))) == NULL)
+        return (dir_store_err(FS2DS(file_system), session, ENOMEM, "dir_store_remove"));
+    if (snprintf(cache_filename, cache_file_len, "%s/%s", dir_store_fs->cache_dir, name) >=
+      (int)cache_file_len)
+        return (dir_store_err(FS2DS(file_system), session, EINVAL, "overflow sprintf"));
+
+    dir_store_exist(file_system, session, name, &existp);
+    if (existp) {
+        ret = wt_fs->fs_remove(wt_fs, session, bucket_filename, flags);
+        if (ret != 0 && ret != ENOENT)
+            goto err;
```

Comment:
Might be good to call `dir_store_err` before the jump to `err` here, and pass it `bucket_filename`.  That way there will be an error message telling which file couldn't be removed.  Same thing after next remove call.